### PR TITLE
Update release schedule documentation

### PR DESCRIPTION
**Pull request by mrjoro**
Updates the release schedule documentation:
- add a new section on the release cadence (for when people ask why we can't push AMP more frequently)
- change first section to more clearly say how long it takes for a change to typically be live for all users; the detailed schedule is moved to later
- other minor style changes

/cc @cramforce @rsimha-amp @choumx

### CODE REVIEW

File: contributing/release-schedule.md [hunk 1]
```diff
@@ -1,32 +1,47 @@
...
 
-### AMP Dev Channel
+We are intentionally cautious with our release cadence.
 
-AMP Dev Channel is a way to opt a browser into using a newer version of the AMP JS libraries.
+In determining how often we should push new versions of AMP to everyone we have to weigh many factors including:
 
-This release **may be less stable** and it may contain features not available to all users. Opt into this option if you'd like to help test new versions of AMP, report bugs or build documents that require a new feature that is not yet available to everyone.
+- stability for the millions of sites/billions of pages built using AMP
+- the cache busting that happens we push a new version
```
Comment (bpaduch) [inline contributing/release-schedule.md]: add "when" before "we"

File: contributing/release-schedule.md [hunk 2]
```diff
@@ -1,32 +1,47 @@
...
+Announcements regarding releases will be made on the [AMP Slack #release channel](https://amphtml.slack.com/messages/C4NVAR0H3/) ([sign up for Slack](https://docs.google.com/forms/d/e/1FAIpQLSd83J2IZA6cdR6jPwABGsJE8YL4pkypAbKMGgUZZriU7Qu6Tg/viewform?fbzx=4406980310789882877)).
 
-Announces regarding releases will be made on the [AMP Slack #release channel](https://amphtml.slack.com/messages/C4NVAR0H3/).  (You will need to [sign up for access](https://docs.google.com/forms/d/e/1FAIpQLSd83J2IZA6cdR6jPwABGsJE8YL4pkypAbKMGgUZZriU7Qu6Tg/viewform?fbzx=4406980310789882877) to AMP on Slack if you haven't done so already].)
+## Release cadence
 
-### AMP Dev Channel
+We are intentionally cautious with our release cadence.
 
-AMP Dev Channel is a way to opt a browser into using a newer version of the AMP JS libraries.
+In determining how often we should push new versions of AMP to everyone we have to weigh many factors including:
```
Comment (bpaduch) [inline contributing/release-schedule.md]: comma after "everyone"

File: contributing/release-schedule.md (hunk 1 above)
Comment (bpaduch) [inline contributing/release-schedule.md]: btw, does cache busting happen every time? If not, consider rewriting as "cache busting that might happen when we push a new version"

File: contributing/release-schedule.md [hunk 3]
```diff
@@ -1,32 +1,47 @@
...
+- Tuesday (about a week after the canary build was cut): the canary is pushed to all users
+
+## AMP Dev Channel
+
+AMP Dev Channel is a way to opt a browser into using a newer version of the AMP JS libraries.  This release **may be less stable** and it may contain features not available to all users. 
+
+Opting into the Dev Channel is great to:
+
+- test and play with new features not yet available to all users
+- use in Q&A to ensure that your site is compatible with the next version of AMP
```
Comment (bpaduch) [inline contributing/release-schedule.md]: change Q&A to Quality Assurance (QA)

File: contributing/release-schedule.md (hunk 2 above)
Comment (mrjoro) [inline contributing/release-schedule.md]: done

File: contributing/release-schedule.md (hunk 1 above)
Comment (mrjoro) [inline contributing/release-schedule.md]: done

File: contributing/release-schedule.md (hunk 3 above)
Comment (mrjoro) [inline contributing/release-schedule.md]: ha, I didn't even notice that :)

**Pull request merged**